woocommerce/woocommerce
# Add selection to the tree control

**Depends on #36590**


### All Submissions:






### Changes proposed in this Pull Request:



Depends on https://github.com/woocommerce/woocommerce/pull/36476
Partialy closes https://github.com/woocommerce/woocommerce/issues/35851






### How to test the changes in this Pull Request:



1. Run storybook
2. Visit `http://localhost:6007/?path=/story/woocommerce-admin-experimental-treecontrol--selection-single` for a single selection tree. Or `http://localhost:6007/?path=/story/woocommerce-admin-experimental-treecontrol--selection-multiple` for a multiple selection tree.
3. Both examples have an initial value set.
4. After clicking a leaf node in the single selection tree, the selected value should be change.
5. In multiple selection tree we can select any children.
5.1. When selecting a leaf, all its ancestors should be selected also.
5.2 When selecting an item with children, the item and its children should be selected.
5.3 When selecting an item with parent and children, its parent and its children should be selected.
5.4 When unselecting a leaf, if it has selected siblings, then it becomes unselected and all its ancestors indeterminate.
5.5 When unselecting a leaf, if it has unselected siblings, then it becomes unselected and all its ancestors unselected.
5.6 When unselecting an item with children, all it children becomes unselected.



### Other information:




### FOR PR REVIEWER ONLY:
